### ISSUE TITLE
Crash on R2 , this is after a few mins of play

### ISSUE THREAD

**Issue by FallenAngelX** [ike3/mangosbot#25]
Revision: 2014-10-14 13:17:12 12687 c0ade47566dbc4c02f64926a32a8e3085182d377
Date 14:10:2014. Time 20:55 
//=====================================================
**\* Hardware ***
Processor: Intel(R) Core(TM) i7-4700MQ CPU @ 2.40GHz
Number Of Processors: 8
Physical Memory: 4194303 KB (Available: 3920380 KB)
Commit Charge Limit: 4194303 KB

**\* Operation System ***
Windows 8 (Version 6.2, Build 9200)

//=====================================================
Exception code: C0000005 ACCESS_VIOLATION
Fault address:  009D48E5 01:005D38E5 C:\Users\fallenangel\Desktop\Testing_Server\MaNGOS\mangosd.exe

Registers:
EAX:2706E4E0
EBX:00000000
ECX:00000002
EDX:00000001
ESI:00000000
EDI:1C6DF020
CS:EIP:0023:009D48E5
SS:ESP:002B:1812C3E0  EBP:1812C76C
DS:002B  ES:002B  FS:0053  GS:002B
Flags:00010246

Call stack:
Address   Frame     Function      SourceFile
009D48E5  00000000  BIH::intersectRayVMAP::MapRayCallback+3A5  c:\users\fallenangel\desktop\mangos\src\game\vmap\bih.h line 192
009D78D3  00000000  VMAP::StaticMapTree::getIntersectionTime+33  c:\users\fallenangel\desktop\mangos\src\game\vmap\maptree.cpp line 147
009D813F  00000000  VMAP::StaticMapTree::isInLineOfSight+1CF  c:\users\fallenangel\desktop\mangos\src\game\vmap\maptree.cpp line 167
00953EF2  00000000  VMAP::VMapManager2::isInLineOfSight+D2  c:\users\fallenangel\desktop\mangos\src\game\vmap\vmapmanager2.cpp line 156
0050E833  00000000  Map::IsInLineOfSight+43  c:\users\fallenangel\desktop\mangos\src\game\map.cpp line 2661
004D0A44  00000000  WorldObject::IsWithinLOS+74  c:\users\fallenangel\desktop\mangos\src\game\object.cpp line 1295
004D0AB8  00000000  WorldObject::IsWithinLOSInMap+68  c:\users\fallenangel\desktop\mangos\src\game\object.cpp line 1288
00AFA648  00000000  ai::NearestGameObjects::Calculate+128  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\strategy\values\nearestgameobjects.cpp line 42
00A67C1B  00000000  ai::CalculatedValuestd::list<ObjectGuid,std::allocator<ObjectGuid > >::Get+4B  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\strategy\value.h line 41
00ADC8D6  00000000  ai::AddAllLootAction::Execute+66  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\strategy\actions\addlootaction.cpp line 23
00A2F60A  00000000  ai::Engine::ListenAndExecute+8A  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\strategy\engine.cpp line 466
00A2E798  00000000  ai::Engine::DoNextAction+348  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\strategy\engine.cpp line 171
00A0A830  00000000  PlayerbotAI::DoNextAction+80  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\playerbotai.cpp line 422
00A0DFEE  00000000  PlayerbotAI::UpdateAIInternal+38E  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\playerbotai.cpp line 174
009F10A0  00000000  PlayerbotAIBase::UpdateAI+30  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\playerbotaibase.cpp line 23
00A0DC3F  00000000  PlayerbotAI::UpdateAI+11F  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\playerbotai.cpp line 150
006067A3  00000000  Player::Update+7B3  c:\users\fallenangel\desktop\mangos\src\game\player.cpp line 1536
004D3A6B  00000000  WorldObject::UpdateHelper::Update+8B  c:\users\fallenangel\desktop\mangos\src\game\object.cpp line 2370
00512694  00000000  Map::Update+404  c:\users\fallenangel\desktop\mangos\src\game\map.cpp line 654
006CA0CF  00000000  ObjectUpdateRequest<Map>::call+F  c:\users\fallenangel\desktop\mangos\src\shared\objectupdatetaskbase.h line 50
006CB04A  00000000  ObjectUpdateTaskBase<Map>::svc+8A  c:\users\fallenangel\desktop\mangos\src\shared\objectupdatetaskbase.h line 102
689AA1BC  00000000  ACE_Task_Base::svc_run+2C  c:\users\fallenangel\desktop\mangos\dep\ace_wrappers\ace\task.cpp line 286
689AA566  00000000  ACE_Thread_Adapter::invoke_i+76  c:\users\fallenangel\desktop\mangos\dep\ace_wrappers\ace\thread_adapter.cpp line 145
689AA4BE  00000000  ACE_Thread_Adapter::invoke+7E  c:\users\fallenangel\desktop\mangos\dep\ace_wrappers\ace\thread_adapter.cpp line 94
68D4C01D  00000000  __get_tlsindex+6F
68D4C001  00000000  __get_tlsindex+53
76F686E3  00000000  BaseThreadInitThunk+E
778DBE99  00000000  RtlInitializeExceptionChain+85
# 778DBE6C  00000000  RtlInitializeExceptionChain+58

Local Variables And Parameters

Call stack:
Address   Frame     Function      SourceFile
009D48E5  00000000  BIH::intersectRayVMAP::MapRayCallback+3A5  c:\users\fallenangel\desktop\mangos\src\game\vmap\bih.h line 192
    Local  <user defined> 'r' = 1812C76C
    Local  <user defined> 'intersectCallback' = 1812C76C
    Local  <user defined> 'maxDist' = 1812C76C
    Local  bool 'stopAtFirst' = 90
    Local  <user defined> 'dir' = 1812C76C
    Local  <user defined> 'stack' = 1812C76C
    Local  <user defined> 'offsetFront' = 1812C76C
    Local  <user defined> 'offsetBack' = 1812C76C
    Local  <user defined> 'offsetFront3' = 1812C76C
    Local  float 'intervalMax' = 0.000000
    Local  float 'intervalMin' = 0.000000
    Local  <user defined> 'org' = 1812C76C
    Local  <user defined> 'invDir' = 1812C76C
    Local  <user defined> 'offsetBack3' = 1812C76C
    Local  float 't1' = 0.000000
    Local  float 't2' = 0.000000
    Local  float 'tf' = 0.000000
    Local  float 'tb' = 0.000000
    Local  float 'tf' = 0.000000
    Local  float 'tb' = 0.000000

009D78D3  00000000  VMAP::StaticMapTree::getIntersectionTime+33  c:\users\fallenangel\desktop\mangos\src\game\vmap\maptree.cpp line 147
    Local  <user defined> 'pRay' = 1812C790
    Local  <user defined> 'pMaxDist' = 1812C790
    Local  bool 'pStopAtFirstHit' = 30
    Local  <user defined> 'intersectionCallBack' = 1812C790
    Local  float 'distance' = 0.000000

009D813F  00000000  VMAP::StaticMapTree::isInLineOfSight+1CF  c:\users\fallenangel\desktop\mangos\src\game\vmap\maptree.cpp line 167
    Local  <user defined> 'pos1' = 1812D830
    Local  <user defined> 'pos2' = 1812D830
    Local  <user defined> 'ray' = 1812D830
    Local  float 'maxDist' = 0.000000
    Local  <user defined> 'st' = 1812D830

00953EF2  00000000  VMAP::VMapManager2::isInLineOfSight+D2  c:\users\fallenangel\desktop\mangos\src\game\vmap\vmapmanager2.cpp line 156
    Local  unsigned int 'pMapId' = 1812D894
    Local  float 'x1' = 0.000000
    Local  float 'y1' = 0.000000
    Local  float 'z1' = 0.000000
    Local  float 'x2' = 0.000000
    Local  float 'y2' = 0.000000
    Local  float 'z2' = 0.000000
    Local  <user defined> 'instanceTree' = 1812D86C
    Local  <user defined> 'pos1' = 1812D86C
    Local  <user defined> 'pos2' = 1812D86C

0050E833  00000000  Map::IsInLineOfSight+43  c:\users\fallenangel\desktop\mangos\src\game\map.cpp line 2661
    Local  float 'srcX' = 0.000000
    Local  float 'srcY' = 0.000000
    Local  float 'srcZ' = 0.000000
    Local  float 'destX' = 0.000000
    Local  float 'destY' = 0.000000
    Local  float 'destZ' = 0.000000
    Local  unsigned int 'phasemask' = 1812D8C4

004D0A44  00000000  WorldObject::IsWithinLOS+74  c:\users\fallenangel\desktop\mangos\src\game\object.cpp line 1295
    Local  float 'ox' = 0.000000
    Local  float 'oy' = 0.000000
    Local  float 'oz' = 0.000000
    Local  float 'z' = 0.000000
    Local  float 'x' = 0.000000
    Local  float 'y' = 0.000000

004D0AB8  00000000  WorldObject::IsWithinLOSInMap+68  c:\users\fallenangel\desktop\mangos\src\game\object.cpp line 1288
    Local  <user defined> 'obj' = 1812D8E0
    Local  float 'oz' = 0.000000
    Local  float 'oy' = 0.000000
    Local  float 'ox' = 0.000000

00AFA648  00000000  ai::NearestGameObjects::Calculate+128  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\strategy\values\nearestgameobjects.cpp line 42
    Local  <user defined> 'targets' = 1812E93C
    Local  <user defined> 'u_check' = 1812E93C
    Local  <user defined> 'searcher' = 1812E93C

00A67C1B  00000000  ai::CalculatedValuestd::list<ObjectGuid,std::allocator<ObjectGuid > >::Get+4B  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\strategy\value.h line 41

00ADC8D6  00000000  ai::AddAllLootAction::Execute+66  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\strategy\actions\addlootaction.cpp line 23
    Local  <user defined> 'event' = 1812E9BC
    Local  <user defined> 'gos' = 1812E9BC
    Local  <user defined> 'corpses' = 1812E9BC

00A2F60A  00000000  ai::Engine::ListenAndExecute+8A  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\strategy\engine.cpp line 466
    Local  <user defined> 'action' = 1812EA48
    Local  <user defined> 'event' = 1812EA48
    Local  bool 'actionExecuted' = A8

00A2E798  00000000  ai::Engine::DoNextAction+348  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\strategy\engine.cpp line 171
    Local  <user defined> 'unit' = 1812EBA8
    Local  int 'depth' = 1812EC00
    Local  <user defined> 'event' = 1812EBA8
    Local  float 'relevance' = 0.000000

00A0A830  00000000  PlayerbotAI::DoNextAction+80  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\playerbotai.cpp line 422
    Local  <user defined> 'packet' = 1812EC00

00A0DFEE  00000000  PlayerbotAI::UpdateAIInternal+38E  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\playerbotai.cpp line 174
    Local  unsigned int 'elapsed' = 1812ED34
    Local  <user defined> 'helper' = 1812ED20
    Local  <user defined> 'holder' = 1812ED20
    Local  <user defined> 'command' = 1812ED20
    Local  <user defined> 'out' = 1812ED20

009F10A0  00000000  PlayerbotAIBase::UpdateAI+30  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\playerbotaibase.cpp line 23
    Local  unsigned int 'elapsed' = 1812ED5C

00A0DC3F  00000000  PlayerbotAI::UpdateAI+11F  c:\users\fallenangel\desktop\mangos\src\plugins\playerbot\playerbotai.cpp line 150
    Local  unsigned int 'elapsed' = 1812EDD0

006067A3  00000000  Player::Update+7B3  c:\users\fallenangel\desktop\mangos\src\game\player.cpp line 1536
    Local  unsigned int 'update_diff' = 1812EDF4
    Local  unsigned int 'p_time' = 1812EDF4
    Local  <user defined> 'iter' = 1812EDD0
    Local  float 'bubble' = 0.000000
    Local  unsigned int 'newzone' = 1812EDF4
    Local  unsigned int 'newarea' = 1812EDF4
    Local  unsigned int 'timer' = 1812EDF4

004D3A6B  00000000  WorldObject::UpdateHelper::Update+8B  c:\users\fallenangel\desktop\mangos\src\game\object.cpp line 2370
    Local  unsigned int 'time_diff' = 1812FE90

00512694  00000000  Map::Update+404  c:\users\fallenangel\desktop\mangos\src\game\map.cpp line 654
    Local  <user defined> 't_diff' = 1812FE90
    Local  <user defined> 'updater' = 1812FE90
    Local  <user defined> 'world_object_update' = 1812FE90
    Local  <user defined> 'updateQueue' = 1812FE90
    Local  <user defined> 'grid_object_update' = 1812FE90
    Local  <user defined> 'guid' = 1812FE90
    Local  <user defined> 'updater' = 1812FE90
    Local  <user defined> 'guid' = 1812FE90
    Local  <user defined> 'helper' = 1812FE90
    Local  <user defined> 'helper' = 1812FE90
    Local  <user defined> 'area' = 1812FE90
    Local  <user defined> 'guid' = 1812FE90
    Local  <user defined> 'cell' = 1812FE90
    Local  <user defined> 'st' = 1812FE90

006CA0CF  00000000  ObjectUpdateRequest<Map>::call+F  c:\users\fallenangel\desktop\mangos\src\shared\objectupdatetaskbase.h line 50

006CB04A  00000000  ObjectUpdateTaskBase<Map>::svc+8A  c:\users\fallenangel\desktop\mangos\src\shared\objectupdatetaskbase.h line 102

689AA1BC  00000000  ACE_Task_Base::svc_run+2C  c:\users\fallenangel\desktop\mangos\dep\ace_wrappers\ace\task.cpp line 286
    Local  args& = 1812FF1C void 'args' = 1812FF1C

689AA566  00000000  ACE_Thread_Adapter::invoke_i+76  c:\users\fallenangel\desktop\mangos\dep\ace_wrappers\ace\thread_adapter.cpp line 145
    Local  unsigned long 'status' = 1812FF40

689AA4BE  00000000  ACE_Thread_Adapter::invoke+7E  c:\users\fallenangel\desktop\mangos\dep\ace_wrappers\ace\thread_adapter.cpp line 94
    Local  <user defined> 'exit_hook_maybe' = 1812FF40

68D4C01D  00000000  __get_tlsindex+6F

68D4C001  00000000  __get_tlsindex+53

76F686E3  00000000  BaseThreadInitThunk+E

778DBE99  00000000  RtlInitializeExceptionChain+85

778DBE6C  00000000  RtlInitializeExceptionChain+58
#

**Comment by ike3** [comment]
Seems bot wanted to loot a gameobject (Herb, Mineral vein or so) but LoS calculation in R2 crashes internally.
Don't think it can be fixed in playerbot as it expects that LoS should be calculated for any spawned gameobject. Cannot reproduce it locally though, could be some VMap issue.

**Comment by FallenAngelX** [comment]
ya ive traced it down, seems R2 is junk anymore unless you revert to a thousand commits ago lol, still has that  crash in ensuregridload. any plans on maybe implementing this to cmangos ?

**Comment by ike3** [comment]
I thought that CMangos was down too) Actually I dropped (c)mangos support when they decided not to implement Dungeon Finder and later switch to boost convinced me to move on to tc.

**Comment by FallenAngelX** [comment]
actually i think its just Xfurry doing cmangos anymore, TC is the best way to go, and still developing strong, but i personally favor and like mangos and have for yrs i like the coding style, easy use and ect ect, im working to implement your server bots to my current projects soon https://github.com/Infinity-MaNGOS , really excited about it

**Comment by Lillecarl** [comment]
CMaNGOS got some things going, there are changes coming. And they're also working on boost ( https://github.com/qvipka/RustEmu )

TC is good, but i find working with mangos easier, just wanted to share my opinion too!

CMaNGOS is also nice because it supports 1.12.1/2, 2.4.3, 3.3.5 and maybe some time 4.3.4 :)

**Comment by FallenAngelX** [comment]
i miss the old mangos crew , and my old dev team on mangosR2,  mangos needs to focus on spells and LFG instead of boost  i mean theres so much more that needs to be done. me and ike3 where just talking about cmangos going to boost and not doing LFG so he drop support of his serverbot for mangos and moved to mangos, alot of good people have come and gone 8(

**Comment by Lillecarl** [comment]
Yes i remember when i first joined mangos, then we had some great activity, but it started slowing down sadly.

Then again, if you were to start work on LFG and was serious about it (or even if MR2 LFG is good enough to just port over pretty much). Then that would probably get into master, we can't always expect others to do the work we want them to, this is free work after all =)

TL;DR: TrinityCore for 3.3.5 is unbeatable, but then you limit yourself to 3.3.5 and 4.3.4 (which for me would never work)

**Comment by FallenAngelX** [comment]
MR2 lfg was ported over for from trinity lol , ive actually thought about trying to port it over to cmangos but like R2 better lol i just spent the last 2 days getting and fixing up a stable version for windows and rebuilding  vc files back together for 
 https://github.com/Infinity-MaNGOS   for my own R2 repo

**Comment by kvipka** [comment]
funny =)
Infinity-420, RustEmu now more R2 then cmangos =) With LFG and etc. + On boost, yes )
https://github.com/healthstone/RustEmu-Core
